dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityChangeLogService.java

```diff
@@ -97,10 +104,14 @@ public Page<TrackedEntityChangeLog> getTrackedEntityChangeLog(
     TrackedEntity trackedEntity =
         trackedEntityService.getTrackedEntity(
             trackedEntityUid, programUid, TrackedEntityParams.FALSE.withIncludeAttributes(true));
+    Program program =
+        (programUid != null) ? programService.getProgram(programUid.getValue()) : null;
 
     Set<UID> trackedEntityAttributes =
         trackedEntityAttributeService
-            .getTrackedEntityTypeAttributes(trackedEntity.getTrackedEntityType())
+            .getAllUserReadableTrackedEntityAttributes(
```

Comment:
`trackedEntityService.getTrackedEntity(
            trackedEntityUid, programUid, TrackedEntityParams.FALSE.withIncludeAttributes(true));`
is getting the tracked entity and all the tracked entity type attributes and program attributes, should we simply get the attributes with `trackedEntity.trackedEntityAttributeValues()`?